### PR TITLE
Fixed actual plan not working for some queries

### PR DESCRIPTION
**Pull request by laurennat**
Fixes https://github.com/microsoft/vscode-mssql/issues/18376
actualPlanQueryBugFix

### CODE REVIEW

File: src/queryResult/queryResultWebViewController.ts [hunk 1]
```diff
@@ -336,4 +336,8 @@ export class QueryResultWebviewController extends ReactWebviewViewController<
         });
         return total;
     }
+
+    public recordLength(record: any): number {
```
Comment (Benjin) [inline src/queryResult/queryResultWebViewController.ts:340]: Move this to `utils`

File: src/queryResult/utils.ts [hunk 2]
```diff
@@ -86,16 +86,36 @@ export function registerCommonRequestHandlers(
...
+                    1 &&
+            message.resultId ===
+                webviewViewController.recordLength(
+                    currentState.resultSetSummaries[message.batchId],
+                ) -
+                    1 &&
+            (!currentState.executionPlanState?.xmlPlans ||
+                !webviewViewController.recordLength(
+                    currentState.executionPlanState.xmlPlans,
+                ))
```
Comment (Benjin) [inline src/queryResult/utils.ts:115]: Thanks for adding the comment block at the top!  For cascading boolean conditions like this one, I think it's _also_ nice to have comments on each "clause" in the conditional.

Might be worth checking to see if adding more parens tells Prettier to format it more nicely (it's weird that the `1` is on a different line from the measure it's being subtracted from)

Comment (laurennat) [reply src/queryResult/utils.ts]: prettier removes "unnecessary" parens, so that's a no go

Comment (laurennat) [reply src/queryResult/utils.ts]: for what it's worth, I do think moving the recordLength function to utils helps

**Review by Benjin — APPROVED** [review]

**Pull request merged**